src/services/storage/secureStorage.ts

```diff
@@ -0,0 +1,194 @@
+    key: string,
+    options?: SecureStorageOptions,
+  ): Promise<Keychain.UserCredentials | false> => {
+    try {
+      // If explicit biometric prompt is provided, show it first
+      if (options?.explicitBiometricPrompt) {
+        const hasVerified = await rnBiometrics.simplePrompt({
+          promptMessage: options.explicitBiometricPrompt.title,
+          cancelButtonText: options.explicitBiometricPrompt.cancel,
+        });
+        if (!hasVerified.success) {
+          return false;
+        }
+      }
+
+      const result = await Keychain.getGenericPassword({
+        service: `${serviceName}_${key}`,
+        accessControl: SECURE_KEYCHAIN_OPTIONS.accessControl,
+      });
+
+      return result;
+    } catch (error) {
+      logger.error(
+        "secureStorage.getItem",
+        `Error retrieving key from keychain: ${key}`,
+        error,
+      );
+      return false;
+    }
+  },
+
+  /**
+   * Removes one or more items from secure storage
+   *
+   * @param keys - The key(s) to remove
+   */
+  remove: async (keys: string | string[]): Promise<void> => {
+    try {
+      if (Array.isArray(keys)) {
+        await Promise.all(
+          keys.map((key) =>
+            Keychain.resetGenericPassword({
+              service: `${serviceName}_${key}`,
+            }),
+          ),
+        );
+        return;
+      }
+
+      await Keychain.resetGenericPassword({
+        service: `${serviceName}_${keys}`,
+      });
+    } catch (error) {
+      logger.error(
+        "secureStorage.remove",
+        "Error removing keys from keychain",
+        error,
+      );
+      // Don't throw since removal failures shouldn't block execution
+    }
+  },
+
+  /**
+   * Checks if an item exists in secure storage
+   *
+   * @param key - The storage key to check
+   * @returns True if the item exists, false otherwise
+   */
+  checkIfExists: async (key: string): Promise<boolean> => {
+    try {
+      const result = await Keychain.hasGenericPassword({
+        service: `${serviceName}_${key}`,
+      });
+      return result;
+    } catch (error) {
+      logger.error(
+        "secureStorage.checkIfExists",
+        `Error checking if key exists: ${key}`,
+        error,
+      );
+      return false;
+    }
+  },
+
+  /**
+   * Clears all secure storage
+   *
+   * Note: This uses the service name only, which may not match all individual entries
+   * depending on how they were stored. For more reliable clearing, use remove() with
+   * specific keys.
+   */
+  clear: async (): Promise<void> => {
+    try {
+      await Keychain.resetGenericPassword({
+        service: serviceName,
+      });
+    } catch (error) {
+      logger.error("secureStorage.clear", "Error clearing keychain", error);
+    }
+  },
+});
+
+/**
+ * Default secure storage instance using freighter_secure_storage service name
+ * Maintains backward compatibility with existing data
+ */
+export const secureStorage = createSecureStorage(SECURE_STORAGE_SERVICE);
```

Comment:
it looks like we are not using this export. If so, can we remove it to mitigate confusion? Thanks